cli/src/main/java/com/box/l10n/mojito/cli/command/PullCommand.java

```diff
@@ -496,6 +594,45 @@ LocalizedAssetBody getLocalizedAssetBodySync(
     return localizedAsset;
   }
 
+  Map<String, LocalizedAssetBody> getLocalizedAssetBodyParallel(
+      FileMatch sourceFileMatch,
+      List<RepositoryLocale> repositoryLocales,
+      Map<RepositoryLocale, List<String>> localeIdToOutputTagsMap,
+      List<String> filterOptions,
+      Asset assetByPathAndRepositoryId,
+      String assetContent)
+      throws CommandException {
+    Map<String, LocalizedAssetBody> localizedAssetBodyMap = new HashMap<>();
+    MultiLocalizedAssetBody multiLocalizedAsset;
+
+    PollableTask localizedAssetForContentParallel =
+        assetClient.getLocalizedAssetForContentParallel(
+            assetByPathAndRepositoryId.getId(),
+            assetContent,
+            repositoryLocales,
+            localeIdToOutputTagsMap,
+            sourceFileMatch.getFileType().getFilterConfigIdOverride(),
+            filterOptions,
+            status,
+            inheritanceMode,
+            pullRunName);
+    commandHelper.waitForPollableTask(localizedAssetForContentParallel.getId());
```

Comment:
waitForPollableTask can take a WaitForPollableTaskListener. if there enough information in the pollabletask, it could be used to display an improved status update in the console